src/item_factory.cpp

```diff
@@ -1932,15 +1932,32 @@ bool load_min_max(std::pair<T, T> &pa, JsonObject &obj, const std::string &name)
     return result;
 }
 
-bool load_sub_ref(std::unique_ptr<Item_spawn_data> &ptr, JsonObject &obj, const std::string &name)
+bool Item_factory::load_sub_ref( std::unique_ptr<Item_spawn_data> &ptr, JsonObject &obj,
+                                 const std::string &name, const Item_group &parent )
 {
     if (obj.has_member(name)) {
         // TODO!
-    } else if (obj.has_member(name + "-item")) {
+    } else if( obj.has_array( name + "-item" ) || obj.has_array( name + "-group" ) ) {
```

Comment:
These two concatenations were used quite alot. It would be cleaner to use temp. variables instead:
```c++
const std::string item_name( name + "-item" );
const std::string group_name( name + "-group" );

```